components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java

```diff
@@ -268,26 +271,31 @@ public void setResponseHandler(@Nonnull ResponseHandler responseHandler) {
     /**
      * Sets the request body to be a binary stream.
      * @param value the binary stream
+     * @throws IOException when the stream cannot be read.
      * @deprecated use {@link #setStreamContent(InputStream, String)} instead.
      */
     @Deprecated
-    public void setStreamContent(@Nonnull final InputStream value) {
+    public void setStreamContent(@Nonnull final InputStream value) throws IOException {
         setStreamContent(value, BINARY_CONTENT_TYPE);
     }
 
     /**
      * Sets the request body to be a binary stream.
      * @param value the binary stream
      * @param contentType the content type of the stream.
+     * @throws IOException when the stream cannot be read.
      */
     public void setStreamContent(
-            @Nonnull final InputStream value, @Nonnull final String contentType) {
+            @Nonnull final InputStream value, @Nonnull final String contentType)
+            throws IOException {
         Objects.requireNonNull(value);
         Objects.requireNonNull(contentType);
         if (contentType.isEmpty()) {
             throw new IllegalArgumentException("contentType cannot be empty");
         }
         this.content = value;
+        this.contentLength = (long) value.available();
```

Comment:
I'm not sure this is correct, since this is public API and we are not in control of the argument `available` might have very different meanings:

https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#available--

Reply:
☝️